File: matiasmasso/Mat.NET/DTO/DTOTransmisio.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOTransmisio : DTOBaseGuid
    {
        public class Compact
        {
            public Guid Guid { get; set; }
            public int Id { get; set; }
        }
        public DTOEmp Emp { get; set; }
        public int Id { get; set; }
        public DateTimeOffset Fch { get; set; }
        public DTOMgz Mgz { get; set; }
        public DTOAmt Amt { get; set; }
        public List<DTODelivery> Deliveries { get; set; }

        public int DeliveriesCount { get; set; }
        public int InvoicedDeliveriesCount { get; set; }
        public int NoFacturablesCount { get; set; }
        public int LinesCount { get; set; }
        public int UnitsCount { get; set; }

        public DTOTransmisio() : base()
        {
        }

        public DTOTransmisio(Guid oGuid) : base(oGuid)
        {
        }

        public static DTOTransmisio Factory(DTOEmp oEmp, DTOMgz oMgz, List<DTODelivery> oDeliveries)
        {
            DTOTransmisio retval = new DTOTransmisio();
            {
                var withBlock = retval;
                withBlock.Emp = oEmp;
                withBlock.Fch = DateTimeOffset.Now;
                withBlock.Mgz = oEmp.Mgz;
                withBlock.Deliveries = oDeliveries;
            }
            return retval;
        }

        public string FileNameDades()
        {
            string s = FilePrefix() + "dades." + base.Guid.ToString() + ".xml";
            return s;
        }

        public string FileNameDocs()
        {
            return FilePrefix() + "documentacio." + base.Guid.ToString() + ".pdf";
        }

        public string FilePrefix()
        {
            return "M+O." + Fch.Year.ToString() + "." + String.Format("{0:0000}", Id) + ".";
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOTransmisio.vb
Public Class DTOTransmisio
    Inherits DTOBaseGuid

    Property emp As DTOEmp
    Property id As Integer
    Property fch As DateTimeOffset
    Property mgz As DTOMgz
    Property amt As DTOAmt
    Property deliveries As List(Of DTODelivery)

    Property deliveriesCount As Integer
    Property invoicedDeliveriesCount As Integer
    Property noFacturablesCount As Integer
    Property linesCount As Integer
    Property unitsCount As Integer

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function Factory(oEmp As DTOEmp, oMgz As DTOMgz, oDeliveries As List(Of DTODelivery)) As DTOTransmisio
        Dim retval As New DTOTransmisio
        With retval
            .Emp = oEmp
            .Fch = DateTimeOffset.Now
            .Mgz = oEmp.Mgz
            .Deliveries = oDeliveries
        End With
        Return retval
    End Function

    Public Function FileNameDades() As String
        Dim s As String = FilePrefix() & "dades." & MyBase.Guid.ToString & ".xml"
        Return s
    End Function

    Public Function FileNameDocs() As String
        Return FilePrefix() & "documentacio." & MyBase.Guid.ToString & ".pdf"
    End Function

    Public Function FilePrefix() As String
        Return "M+O." & _Fch.Year.ToString & "." & Format(_Id, "0000") & "."
    End Function
End Class
